### ISSUE TITLE
Conflict between "typedef" and "no-inferrable-types"

### ISSUE THREAD

**Issue by inad9300** [palantir/tslint#3595]
Using this rules
```json
{
    "rules": {
        "no-inferrable-types": true,
        "typedef": [true, "variable-declaration"]
    }
}
```

I get the error "Type number trivially inferred from a number literal, remove type annotation" with the code

```typescript
let i: number = 0
```

And the error "expected variable-declaration: 'i' to have a typedef" with the code

```typescript
let i = 0
```

I would expect that the types are not enforced where they can be inferred, if `no-inferrable-types` is set.

Also, note the inconsistency between error messages: only one of them starts with uppercase.

BTW, is there any chance that `no-inferrable-types` can be improved beyond primitive types? TypeScript type inference is pretty good, and it's a pity not to be able to fully benefit from it and from TSLint at the same time.

**Comment by ajafff** [comment]
Duplicate of #711

> BTW, is there any chance that no-inferrable-types can be improved beyond primitive types? TypeScript type inference is pretty good, and it's a pity not to be able to fully benefit from it and from TSLint at the same time.

This is part of my custom rules package: https://github.com/ajafff/tslint-consistent-codestyle/blob/master/docs/no-unnecessary-type-annotation.md. Any feedback is greatly appreciated.
